test/xpu_api/utilities/optional/optional.object/optional.object.observe/dereference.pass.cpp

```diff
@@ -0,0 +1,98 @@
+// -*- C++ -*-
+//===----------------------------------------------------------------------===//
+//
+// Copyright (C) Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// This file incorporates work covered by the following copyright and permission
+// notice:
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+//
+//===----------------------------------------------------------------------===//
+
+#include "support/test_config.h"
+
+#include <oneapi/dpl/optional>
+#include <oneapi/dpl/type_traits>
+
+#include "support/test_macros.h"
+#include "support/utils.h"
+
+#if TEST_DPCPP_BACKEND_PRESENT
+using dpl::optional;
+
+struct X
+{
+    constexpr int
+    test() const&
+    {
+        return 3;
+    }
+    int
+    test() &
+    {
+        return 4;
+    }
+    constexpr int
+    test() const&&
+    {
+        return 5;
+    }
+    int
+    test() &&
+    {
+        return 6;
+    }
+};
+
+struct Y
+{
+    constexpr int
+    test()
+    {
+        return 7;
+    }
+};
```

Comment:
I propose to better mark all of the overloads in ``X`` as constexpr and test `X` using static assertions instead of `Y`.

Reply:
Which profit we have in this  case?

Reply:
Fixed